### PR TITLE
chore: Better Deprecation Warnings GM-198

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Cleans up deprecation notices on core components.
- For components with no existing new component link to the gamut notion board.
- For components that have a new component link to the storybook for that component.
- For anything else just an expanded warning.

### PR Checklist

- [ ] ~Related to designs:~
- [x] Related to JIRA ticket: https://codecademy.atlassian.net/browse/GM-198
- [x] I have run this code to verify it works
- [ ] ~This PR includes unit tests for the code change~

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.10-alpha.40227a.0<br/>@codecademy/gamut-kit@0.3.6-alpha.40227a.0<br/>@codecademy/gamut-labs@13.3.6-alpha.40227a.0<br/>@codecademy/gamut-styles@9.3.1-alpha.40227a.0<br/>@codecademy/gamut-tests@2.3.7-alpha.40227a.0<br/>@codecademy/gamut@29.9.2-alpha.40227a.0<br/>@codecademy/konamimojisplosion@0.2.7-alpha.40227a.0<br/>@codecademy/markdown-overrides@0.5.20-alpha.40227a.0<br/>@codecademy/styleguide@33.8.2-alpha.40227a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6073cde4c6f45dd6d6b519d0--gamut-preview.netlify.app>https://6073cde4c6f45dd6d6b519d0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6073cde4c6f45dd6d6b519d0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.11-alpha.2e0b5c.0<br/>@codecademy/gamut-kit@0.3.10-alpha.2e0b5c.0<br/>@codecademy/gamut-labs@13.4.2-alpha.2e0b5c.0<br/>@codecademy/gamut-styles@9.3.2-alpha.2e0b5c.0<br/>@codecademy/gamut-tests@2.3.8-alpha.2e0b5c.0<br/>@codecademy/gamut@29.9.4-alpha.2e0b5c.0<br/>@codecademy/konamimojisplosion@0.2.8-alpha.2e0b5c.0<br/>@codecademy/markdown-overrides@0.5.21-alpha.2e0b5c.0<br/>@codecademy/styleguide@33.9.2-alpha.2e0b5c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607454b6b2055b339131fa9f--gamut-preview.netlify.app>https://607454b6b2055b339131fa9f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607454b6b2055b339131fa9f>Deploy Logs</a></p>

File: packages/gamut-styles/src/variables/colors.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ const white = '#ffffff';
 
 /**
  * @deprecated
+ * Using these variables directly is no longer supported.
+ *
+ * Please use [`theme.colors`](https://gamut.codecademy.com/storybook/?path=/docs/foundations-theme--colors#standard-colors)
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/variables/colors.ts:8]: Awesome, TIL markdown links work in JSDocs now!

```suggestion
 * Please use [`theme.colors`](https://gamut.codecademy.com/storybook/?path=/docs/foundations-theme--colors#color)
```

File: packages/gamut/src/ButtonDeprecated/index.tsx [hunk 2]
```diff
@@ -132,6 +132,15 @@ const isPreset = (theme: string): theme is keyof typeof buttonPresetThemes => {
 
 /**
  * @deprecated
+ * This component is deprecated and is no longer supported.
+ *
+ * See [FillButon](https://gamut.codecademy.com/storybook/?path=/docs/atoms-button--fill-button#fill-button)
+ *
+ * ```
+ * import { FillButton } fom '@codecademy/gamut'l
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/ButtonDeprecated/index.tsx:140]:
```suggestion
 * import { FillButton } fom '@codecademy/gamut';
```

File: packages/gamut/src/Accordion/index.tsx [hunk 3]
```diff
@@ -49,6 +49,9 @@ export type AccordionProps = {
 
 /**
  * @deprecated
+ * This component is in the old visual identity and will be updated soon.
+ *
+ * Check the [Gamut Board](https://www.notion.so/codecademy/Gamut-Status-Timeline-dd3c135d3848464ea6eb1b48e68fbb1) for component status
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Accordion/index.tsx:54]: Err, I don't have access to this Notion link. Is it private?

File: packages/gamut/src/Typography/HeadingDeprecated.tsx [hunk 4]
```diff
@@ -11,7 +11,7 @@ export type HeadingTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
 export type FontSizes = 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
 
 export type HeadingDeprecatedProps = {
-  /** Heading tags h1 - h6 */
+  /** @param */
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Typography/HeadingDeprecated.tsx:14]:
```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Awesome, thanks Aaron!!

File: packages/gamut/src/Accordion/index.tsx (hunk 3 above)
Comment (codecaaron) [reply packages/gamut/src/Accordion/index.tsx:54]: Hmm maybe vscode is picking a browser for you strangely?  Should be open to everyone on the notion :/

File: packages/gamut-styles/src/variables/colors.ts (hunk 1 above)
Comment (codecaaron) [reply packages/gamut-styles/src/variables/colors.ts:8]: Heh yeah its the workaround for TS and @link not working apparently.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.11-alpha.03f4b5.0<br/>@codecademy/gamut-kit@0.3.10-alpha.03f4b5.0<br/>@codecademy/gamut-labs@13.4.2-alpha.03f4b5.0<br/>@codecademy/gamut-styles@9.3.2-alpha.03f4b5.0<br/>@codecademy/gamut-tests@2.3.8-alpha.03f4b5.0<br/>@codecademy/gamut@29.9.4-alpha.03f4b5.0<br/>@codecademy/konamimojisplosion@0.2.8-alpha.03f4b5.0<br/>@codecademy/markdown-overrides@0.5.21-alpha.03f4b5.0<br/>@codecademy/styleguide@33.9.2-alpha.03f4b5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60745c5859f5943ce51f4b51--gamut-preview.netlify.app>https://60745c5859f5943ce51f4b51--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60745c5859f5943ce51f4b51>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.11-alpha.920ed4.0<br/>@codecademy/gamut-kit@0.3.10-alpha.920ed4.0<br/>@codecademy/gamut-labs@13.4.2-alpha.920ed4.0<br/>@codecademy/gamut-styles@9.3.2-alpha.920ed4.0<br/>@codecademy/gamut-tests@2.3.8-alpha.920ed4.0<br/>@codecademy/gamut@29.9.4-alpha.920ed4.0<br/>@codecademy/konamimojisplosion@0.2.8-alpha.920ed4.0<br/>@codecademy/markdown-overrides@0.5.21-alpha.920ed4.0<br/>@codecademy/styleguide@33.9.2-alpha.920ed4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6074637f67e72d3c07eb3890--gamut-preview.netlify.app>https://6074637f67e72d3c07eb3890--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6074637f67e72d3c07eb3890>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.9.11-alpha.69bde7.0<br/>@codecademy/gamut-kit@0.3.11-alpha.69bde7.0<br/>@codecademy/gamut-labs@13.4.3-alpha.69bde7.0<br/>@codecademy/gamut-styles@9.3.2-alpha.69bde7.0<br/>@codecademy/gamut-tests@2.3.8-alpha.69bde7.0<br/>@codecademy/gamut@29.9.4-alpha.69bde7.0<br/>@codecademy/konamimojisplosion@0.2.8-alpha.69bde7.0<br/>@codecademy/markdown-overrides@0.5.21-alpha.69bde7.0<br/>@codecademy/styleguide@33.9.3-alpha.69bde7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6074675cb7b268444ea021e7--gamut-preview.netlify.app>https://6074675cb7b268444ea021e7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6074675cb7b268444ea021e7>Deploy Logs</a></p>

**Pull request merged**